pkg/operator/ceph/csi/secrets.go

```diff
@@ -0,0 +1,197 @@
+	csiCephFSProvisionerSecret          = "rook-csi-cephfs-provisioner"
+)
+
+func createCSIKeyringRBDNode(s *keyring.SecretStore) (string, error) {
+	key, err := s.GenerateKey(csiKeyringRBDNodeUsername, cephCSIKeyringRBDNodeCaps())
+	if err != nil {
+		return "", err
+	}
+
+	return key, nil
+}
+
+func createCSIKeyringRBDProvisioner(s *keyring.SecretStore) (string, error) {
+	key, err := s.GenerateKey(csiKeyringRBDProvisionerUsername, cephCSIKeyringRBDProvisionerCaps())
+	if err != nil {
+		return "", err
+	}
+
+	return key, nil
+}
+
+func createCSIKeyringCephFSNode(s *keyring.SecretStore) (string, error) {
+	key, err := s.GenerateKey(csiKeyringCephFSNodeUsername, cephCSIKeyringCephFSNodeCaps())
+	if err != nil {
+		return "", err
+	}
+
+	return key, nil
+}
+
+func createCSIKeyringCephFSProvisioner(s *keyring.SecretStore) (string, error) {
+	key, err := s.GenerateKey(csiKeyringCephFSProvisionerUsername, cephCSIKeyringCephFSProvisionerCaps())
+	if err != nil {
+		return "", err
+	}
+
+	return key, nil
+}
+
+func cephCSIKeyringRBDNodeCaps() []string {
+	return []string{
+		"mon", "profile rbd",
+		"osd", "profile rbd",
+	}
+}
+
+func cephCSIKeyringRBDProvisionerCaps() []string {
+	return []string{
+		"mon", "profile rbd",
+		"mgr", "allow rw",
+		"osd", "profile rbd",
+	}
+}
+
+func cephCSIKeyringCephFSNodeCaps() []string {
+	return []string{
+		"mon", "allow r",
+		"osd", "allow rw tag cephfs data=*",
+		"mds", "allow rw",
+	}
+}
+
+func cephCSIKeyringCephFSProvisionerCaps() []string {
+	return []string{
+		"mon", "allow r",
+		"mgr", "allow rw",
+		"osd", "allow rw tag cephfs meta=*",
+	}
+}
+
+func createOrUpdateCSISecret(namespace, csiRBDProvisionerSecretKey, csiRBDNodeSecretKey, csiCephFSProvisionerSecretKey, csiCephFSNodeSecretKey string, k *keyring.SecretStore) error {
+	csiRBDProvisionerSecrets := map[string][]byte{
+		// userID is expected for the rbd provisioner driver
+		"userID":  []byte("csi-rbd-provisioner"),
+		"userKey": []byte(csiRBDProvisionerSecretKey),
+	}
+
+	csiRBDNodeSecrets := map[string][]byte{
+		// userID is expected for the rbd node driver
+		"userID":  []byte("csi-rbd-node"),
+		"userKey": []byte(csiRBDNodeSecretKey),
+	}
+
+	csiCephFSProvisionerSecrets := map[string][]byte{
+		// adminID is expected for the cephfs provisioner driver
+		"adminID":  []byte("csi-cephfs-provisioner"),
+		"adminKey": []byte(csiCephFSProvisionerSecretKey),
+	}
+
+	csiCephFSNodeSecrets := map[string][]byte{
+		// adminID is expected for the cephfs node driver
+		"adminID":  []byte("csi-cephfs-node"),
+		"adminKey": []byte(csiCephFSNodeSecretKey),
+	}
+
+	keyringSecretMap := make(map[string]map[string][]byte)
+	keyringSecretMap[csiRBDProvisionerSecret] = csiRBDProvisionerSecrets
+	keyringSecretMap[csiRBDNodeSecret] = csiRBDNodeSecrets
+	keyringSecretMap[csiCephFSProvisionerSecret] = csiCephFSProvisionerSecrets
+	keyringSecretMap[csiCephFSNodeSecret] = csiCephFSNodeSecrets
+
+	for secret := range keyringSecretMap {
```

Comment:
nit: you could get the secret data from the iterator:
```
for secretName, secret := range keyringSecretMap {
```

Reply:
to clarify, if you defined the iterator as such, this line below would be:
```
Data: secret,
```
instead of
```
Data: keyringSecretMap[secret],
```

Reply:
Got it!